In [1]:
import os.path as path # used for easily finding the csvs in other directories
import pandas as pd, numpy as np, matplotlib.pyplot as plt, seaborn as sns
from datetime import datetime

In [2]:
# loading csv from local directory function
def load_csv_within_directory(directory_string):
    """
    Takes in a string of where your csv is located in the repo folder and turns
    into a dataframe using pandas read_csv

    example directory string: '/data/raw_data/CIE/client_needs_table.csv'
    :param directory_string: string containing file with directory desired
    :return: a dataframe
    """

    temp_path = path.abspath(path.join("data" ,"../../..")) # finds the parent directory

    temp_directory_end = temp_path + directory_string # concatenates with directory for csvs
    temp_df = pd.read_csv(temp_directory_end) # creates df with csv
    return temp_df

In [12]:
# loading demographics for some basic wrangling
demographics = load_csv_within_directory("/data/raw_data/CIE/client_demographics_table.csv")

#quickly renaming columns
old_column_names = list(demographics.columns.values)

#doing quick list comprehension-based string replacing (blanks to _)
new_column_names = [name.replace(" ", "_") for name in old_column_names]
new_column_names = [name.replace("/", "__") for name in new_column_names]

#converting into quick dictionary for replacing old column names with new column names
rename_dict = {i:j for i,j in zip(old_column_names,new_column_names)}
demographics = demographics.rename(columns = rename_dict)

#converting columns to numeric
for column in demographics.columns:
    try:
        column = column.to_numeric()
    except:
        continue

i = 0 #index counter
for date in demographics.Account_Created_Date: #for each date within the report

    #convert the date to datetime #11/18/18 #remember uppercase is full 4 digits, lowercase is short 2 digit
    datetime_object = datetime.strptime(date, '%m/%d/%y')

    #replacing based on index
    demographics.at[i, 'Account_Created_Date'] = datetime_object
    i+=1
demographics.head()

# wrangling done. NaNs are left in for replication purposes,
# Now proceeding to cleaning/prepping data for modeling

AccountID Account_Created_Date Record_Type CIE_Consent      Zip  \
0  0012M000021P37VQAS  2018-11-18 00:00:00      Client  No Consent  92040.0   
1  0012M000021P9NyQAK  2018-11-18 00:00:00      Client  No Consent  92064.0   
2  0012M000021Pa72QAC  2018-11-19 00:00:00      Client     Consent  92071.0   
3  0012M000021Pa8AQAS  2018-11-19 00:00:00      Client     Consent  92110.0   
4  0012M000021PalPQAS  2018-11-19 00:00:00      Client     Consent  91910.0   

   Neighborhood    HHSA_Region     County Housing_Needs Utilities_Needs  ...  \
0      LAKESIDE           East  San Diego            No             Yes  ...   
1         POWAY   North Inland  San Diego           Yes             Yes  ...   
2        SANTEE           East  San Diego            No             Yes  ...   
3  SD, OLD TOWN  North Central  San Diego            No              No  ...   
4   CHULA VISTA          South  San Diego            No             Yes  ...   

      TANF Rental_Assistance     Military__Veteran      Employment  \
0  Missing           Missing  Not Military/Veteran       Full-Time   
1       No                No  Not Military/Veteran  Unable to work   
2  Missing           Missing  Not Military/Veteran       Full-Time   
3  Missing           Missing  Not Military/Veteran         Missing   
4  Missing           Missing  Not Military/Veteran       Full-Time   

                Education Health_Insurance Health_Insurance_Type  Health_Plan  \
0  Some College No Degree              Yes              Medi-Cal       Molina   
1        Associate Degree              Yes              Medi-Cal        Other   
2      High School Degree              Yes     Employer Provided      Missing   
3                 Missing               No          No Insurance      Missing   
4       Bachelor's Degree              Yes     Employer Provided      Missing   

  Homeless          Housing_Type  
0  Missing               Missing  
1      Yes           Unsheltered  
2  Missing               Missing  
3      Yes  Homeless Unspecified  
4  Missing               Missing  

[5 rows x 45 columns]

In [14]:
monthly_tenants = load_csv_within_directory("/data/raw_data/monthly_tenants_reached/monthly_tenants_reached.csv")
# will most likely need to normalize if logit is desired

org  Sum_tenants_1  \
0                                                ACCE            NaN   
1                                       Casa Familiar           84.0   
2                                  Chicano Federation           65.0   
3                                   Logan Heights CDC            NaN   
4         Refugee Coalition - Haitian Bridge Alliance            NaN   
5                  Refugee Coalition - Horn of Africa            NaN   
6   Refugee Coalition - Karen organization of San ...            NaN   
7   Refugee Coalition - Somali Bantu Community of ...            NaN   
8   Refugee Coalition - South Sudanese Community o...            NaN   
9   Refugee Coalition - Union of Pan Asian Communi...            NaN   
10  Refugee Coalition - United Women of East Afric...            NaN   
11                                              SDRCC            NaN   

    Sum_tenants_2  Sum_tenants_3  Sum_tenants_4  Sum_tenants_5  Sum_tenants_6  \
0             NaN         1868.0         1255.0         1052.0        15924.0   
1           261.0           80.0          279.0          628.0          634.0   
2           311.0          217.0          106.0           95.0           64.0   
3           125.0            NaN            NaN            NaN            NaN   
4             NaN            NaN            NaN            NaN           50.0   
5             NaN           77.0            NaN         1732.0         2513.0   
6             NaN            NaN            NaN           30.0          155.0   
7             NaN            NaN            NaN           25.0           59.0   
8             NaN            NaN            NaN            NaN            NaN   
9             NaN            NaN          751.0            NaN           50.0   
10            NaN          126.0          332.0          369.0          337.0   
11          227.0          160.0         1924.0            NaN          430.0   

    Sum_tenants_7  Sum_tenants_8  Sum_tenants_9  Sum_tenants_10  \
0           834.0         4048.0         2650.0           764.0   
1          1177.0         1034.0          504.0           536.0   
2            78.0           77.0           66.0           137.0   
3             NaN            NaN            NaN             NaN   
4             NaN            NaN            NaN             NaN   
5          2316.0         2042.0         1457.0          1470.0   
6           100.0           90.0          100.0           124.0   
7             NaN           85.0            NaN             NaN   
8             NaN            NaN            NaN            93.0   
9           556.0            NaN            NaN             NaN   
10            NaN          446.0            NaN             NaN   
11            NaN            NaN            NaN             NaN   

    Sum_tenants_11  Sum_tenants_12  
0           3123.0             NaN  
1            329.0           207.0  
2            356.0           315.0  
3              NaN             NaN  
4              NaN             NaN  
5           1168.0          2033.0  
6              NaN             NaN  
7              NaN             NaN  
8              NaN             NaN  
9              NaN             NaN  
10           912.0             NaN  
11             NaN             NaN

Possible Questions:
- How can we see whether legal representation can determine evictions?
- Are the workshops working? How well?

Models Desired Pertaining to Research Questions!
- (Legal) Combining Attendance of Workshops by Month to Dataframe with Monthly 311 Counts?
- Possible Time-Series Logistic Regression With Grouped Demographics/Location? (Change Over Time)
- Supervised Classification Of Housing_Need using High Dimensional Variables (Demographics)?
-

Random Thoughts
- Could we control for population and income by using Monthly Tenants Reached?
    - Possibly the income for the area for each workshop?
    - Do we have more Monthly Count Data (more years)? Otherwise, we may need to justify spreading aggregate
        - Or, scale based on GPS data from Google/Apple Maps? Or Other count-based data

Data Wrangling Tasks Found
1. Expand SDSC Court Cases (Unmasked)/Wrangle More Data
2. Acquire Sheriff Evictions
3. Workshop data needed (registrants, post-survey/overview)

Data Cleaning Tasks Found
1. Dummy/One-Hot All Categoricals
2. NaN check, if categorical, drop the row. If it's numerical, take the average or median
3. Normalize Tenant Counts (Log Transform)
4. Determine IVs to model with

---

Data reporting
Be super precise about the data reflects

-Court data
    - eviction filings
        - can get some additional information
- Sheriff data
    - landlord files efiction report
        - court rules in favor
            - Judge sets a lockout date
                - Landlord has to pay a fee
                -

-Try finding

-Look at Census Tracts for Eviction Data
    -Useful for GIS


-Workshop Registration
    -Person Interfaced with a website/etc
       -They received some amount of info about eviction prevention
        -Not everyone is at risk of eviction
      - Would need to control for time

Do we see any difference between zip codes where people who are registered for eviction prevention and those who were not?


In [6]:
court_data_CL_Masked = load_csv_within_directory("/data/raw_data/public_records_request_19Jan2023/CL_masked.csv")

In [7]:
court_data_CL_Masked

CASE NUMBER        CITY    ZIP  MONTH  YEAR  \
0      37-2018-00000071-CL-UD-CTL   San Diego  92139      1  2018   
1      37-2018-00000103-CL-UD-CTL   San Diego  92110      1  2018   
2      37-2018-00000117-CL-UD-CTL       Vista  92084      1  2018   
3      37-2018-00000125-CL-UD-CTL   SAN DIEGO  92117      1  2018   
4      37-2018-00000137-CL-UD-CTL    Lakeside  92040      1  2018   
...                           ...         ...    ...    ...   ...   
23119  37-2022-00052500-CL-UD-CTL   San Diego  92101     12  2022   
23120  37-2022-00052501-CL-UD-CTL   San Diego  92101     12  2022   
23121  37-2022-00052502-CL-UD-CTL  San Marcos  92069     12  2022   
23122  37-2022-00052503-CL-UD-CTL   Oceanside  92054     12  2022   
23123  37-2022-00052504-CL-UD-CTL      Alpine  91901     12  2022   

                      DISPO TYPE ANSWER MONEY JUDGMENT  DEFAULT  
0        Court-ordered dismissal      Y              N        N  
1        Court-ordered dismissal      Y              N        N  
2          Request for Dismissal      N              N        Y  
3        Court-ordered dismissal      N              N        Y  
4      Default judgment by clerk      N              N        Y  
...                          ...    ...            ...      ...  
23119                        NaN      Y              N     N, N  
23120                        NaN      N              N     N, N  
23121                        NaN      N              N     N, N  
23122                        NaN      Y              N     N, N  
23123                        NaN      N              N  N, N, N  

[23124 rows x 9 columns]